update.go

```
@@ -0,0 +1,186 @@
+		cli.StringFlag{
+			Name:  "memory-reservation",
+			Usage: "Memory reservation or soft_limit (in bytes)",
+		},
+		cli.StringFlag{
+			Name:  "memory-swap",
+			Usage: "Total memory usage (memory + swap); set `-1` to enable unlimited swap",
+		},
+	},
+	Action: func(context *cli.Context) {
+		container, err := getContainer(context)
+		if err != nil {
+			fatal(err)
+		}
+
+		r := specs.Resources{
+			Memory: &specs.Memory{
+				Limit:       u64Ptr(0),
+				Reservation: u64Ptr(0),
+				Swap:        u64Ptr(0),
+				Kernel:      u64Ptr(0),
+			},
+			CPU: &specs.CPU{
+				Shares: u64Ptr(0),
+				Quota:  u64Ptr(0),
+				Period: u64Ptr(0),
+				Cpus:   sPtr(""),
+				Mems:   sPtr(""),
+			},
+			BlockIO: &specs.BlockIO{
+				Weight: u16Ptr(0),
+			},
+		}
+
+		config := container.Config()
+
+		if in := context.String("ressources"); in != "" {
+			var (
+				f   *os.File
+				err error
+			)
+			switch in {
+			case "-":
+				f = os.Stdin
+			default:
+				f, err = os.Open(in)
+				if err != nil {
+					fatal(err)
+				}
+			}
+			err = json.NewDecoder(f).Decode(&r)
+			if err != nil {
+				fatal(err)
+			}
+		} else {
+			if val := context.Int("blkio-weight"); val != 0 {
+				r.BlockIO.Weight = u16Ptr(uint16(val))
+			}
+			if val := context.String("cpuset-cpus"); val != "" {
+				r.CPU.Cpus = &val
+			}
+			if val := context.String("cpuset-mems"); val != "" {
+				r.CPU.Mems = &val
+			}
+
+			for opt, dest := range map[string]*uint64{
+				"cpu-period":         r.CPU.Period,
+				"cpu-quota":          r.CPU.Quota,
+				"cpu-share":          r.CPU.Shares,
+				"kernel-memory":      r.Memory.Kernel,
+				"memory":             r.Memory.Limit,
+				"memory-reservation": r.Memory.Reservation,
+				"memory-swap":        r.Memory.Swap,
+			} {
+				if val := context.String(opt); val != "" {
+					var err error
+					*dest, err = strconv.ParseUint(val, 10, 64)
+					if err != nil {
+						fatal(fmt.Errorf("Invalid value for %s: %s", opt, err))
+					}
+				}
+			}
+		}
+
```

If one of the value is invalid, the 'Set(' call will error out.

If the invalid value is from one of the unsupported options (e.g. DisableOOMKiller), it will just be ignored.

Just to clarify, you want us to error out instead of ignoring the value in case 2?